src/report.rs

```diff
@@ -0,0 +1,90 @@
+use std::{collections::HashSet, io::Write};
+
+use anyhow::Result;
+use thiserror::Error;
+use time::OffsetDateTime;
+
+/// A specific report of a file's checksum results.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct ChecksumReport {
+    file_path: String,
+    matching: bool,
+    got: String,
+}
+
+impl ChecksumReport {
+    pub fn new(file_path: String, matching: bool, got: String) -> Self {
+        Self {
+            file_path,
+            matching,
+            got,
+        }
+    }
+}
+
+/// A report of all the tool's analysis results.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Report {
+    files_checksums: HashSet<ChecksumReport>,
```

Comment:
There seems to be a semi-common subject-verb disagreement in the `file(s)_checksum(s)` field. I'd suggest choosing one or the other for future consistency and maintainence.